### PR TITLE
Backport Django 3.2 upgrade in maple release

### PR DESCRIPTION
**Pull request by iamsobanjaved**
Contains all commits necessary for Django 3.2 upgrade

Tested this on sandbox here: https://mapledj32.sandbox.edx.org

You may need to fake `auth_user` migration if the table contains quite a large number of users. 
Migration to fake: `0066_alter_user_first_name.py`

### Note
This PR also contains `django-oscar` upgrade commits. It has the migration of a few models from one app to a newly created app. This upgrade will create new tables and will eventually delete old ones. Though it contains records of previously sent order-related emails. Currently, that destructive migration is not part of this PR as the revenue squad is currently testing this on stage, so they want a possibility of a rollback if required. You can create that once the testing is fine.

### CODE REVIEW

**Comment by nedbat** [comment]
@regisb @bbrsofiane This is a large PR, but will get Django 3.2 for ecommerce in Maple.

**Comment by nedbat** [comment]
@iamsobanjaved is this still a draft? If so, what do we need to do to get it merged?

**Comment by iamsobanjaved** [comment]
This is ready from my side. It is open for merging. Depends whether community's review needed before merging or not. I will resolve conflicts in few hours

**Comment by regisb** [comment]
The fact that migration `0066_alter_user_first_name.py` needs to be faked is an issue. It is difficult to automate this sort of manual step. Is there any way we can avoid that? If not, why?

**Comment by BbrSofiane** [comment]
> Currently, that destructive migration is not part of this PR as the revenue squad is currently testing this on stage, so they want a possibility of a rollback if required. You can create that once the testing is fine.

What's the status on the destructive migration?

**Comment by awais786** [comment]
> The fact that migration `0066_alter_user_first_name.py` needs to be faked is an issue. It is difficult to automate this sort of manual step. Is there any way we can avoid that? If not, why?

@iamsobanjaved found a way to avoid this migration. He will push the changes tomorrow. 
Override `first_name = models.CharField(_('first name'), max_length=30, blank=True)` field [here](https://github.com/edx/ecommerce/blob/master/ecommerce/core/models.py#L521) will avoid the migration.

**Comment by awais786** [comment]
> > Currently, that destructive migration is not part of this PR as the revenue squad is currently testing this on stage, so they want a possibility of a rollback if required. You can create that once the testing is fine.
> 
> What's the status on the destructive migration?

Owning team is still working on this. We will inform once they run those migrations.

**Comment by iamsobanjaved** [comment]
@regisb I have pushed the change which will avoid migration on an auth user table, so won't need any faking or manual steps during deployment.

**Comment by iamsobanjaved** [comment]
@BbrSofiane as the initial `django-oscar` upgrade is successful so the PR (https://github.com/edx/ecommerce/pull/3597) with that destructive migration is in progress by owning squad

**Comment by BbrSofiane** [comment]
@iamsobanjaved Thanks for the update!

I am in favour of merging this PR tomorrow so we have a bit of time to test before the official release date (Monday 20th). Is there any risks if that destructive migration is not in Maple?

**Comment by iamsobanjaved** [comment]
@BbrSofiane It is safe to merge without that migration as the revenue squad did the same. And also that migration is now merged too, we can cherry-pick that in this PR too.

**Comment by BbrSofiane** [comment]
Sweet! Let us know when it's cherry-picked.

**Comment by iamsobanjaved** [comment]
Just cherry-picked that migration from master

File: requirements/base.txt [hunk 1]
```diff
@@ -89,7 +91,7 @@ defusedxml==0.7.1
     # via
     #   python3-openid
     #   social-auth-core
-django==2.2.25
+django==3.2.9
```
Comment (awais786) [inline requirements/base.txt:94]: django upgraded.

**Review by awais786 — APPROVED** [review]

**Comment by iamsobanjaved** [comment]
Tested this on sandbox and checkout and refund is working as per expectations 

Sandbox: https://maple32.sandbox.edx.org/dashboard

**Comment by iamsobanjaved** [comment]
Do let me know when you want me to merge this

**Comment by BbrSofiane** [comment]
@iamsobanjaved can we merge this now?

**Comment by iamsobanjaved** [comment]
@BbrSofiane just merged, you can test now

**Pull request merged**